R/init.R

```diff
@@ -183,19 +183,17 @@ init <- function(data,
   }
 
   if (inherits(data, "teal_data")) {
-    if (length(teal.data::datanames(data)) == 0) {
-      stop("`data` object has no datanames. Specify `datanames(data)` and try again.")
+    if (length(teal.data::datanames(data)) == 0 && length(ls(data@env)) == 0) {
+      stop("`data` object has no datanames and its environment is empty. Specify `datanames(data)` and try again.")
     }
-
     # in case of teal_data_module this check is postponed to the srv_teal_with_splash
-    is_modules_ok <- check_modules_datanames(modules, teal.data::datanames(data))
+    is_modules_ok <- check_modules_datanames(modules, teal_data_datanames(data))
     if (!isTRUE(is_modules_ok)) {
       logger::log_error(is_modules_ok)
-      checkmate::assert(is_modules_ok, .var.name = "modules")
+      # we allow app to continue if datanames are not available.
```

Comment:
when using `teal_data_datanames` we should stop the app. I don't necessary mean to crash the app but we can't continue loading modules as there are insufficient datanames. 
```suggestion
      checkmate::assert(is_modules_ok, .var.name = "modules") 
      # or
      validate(need(FALSE, is_modules_ok))
```